flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskExecutor.java

```diff
@@ -309,6 +311,25 @@ public TaskExecutor(
 		return CompletableFuture.completedFuture(shuffleEnvironment.getPartitionsOccupyingLocalResources().isEmpty());
 	}
 
+	@Override
+	public CompletableFuture<Collection<Tuple2<String, Long>>> requestLogList(Time timeout) {
+		final String logDir = taskManagerConfiguration.getTaskManagerLogDir();
+		if (logDir != null) {
+			final File[] logFiles = new File(logDir).listFiles();
+
+			if (logFiles == null) {
+				return FutureUtils.completedExceptionally(
+					new FlinkException("The specific log directory is not a valid directory."));
+			}
+
+			final List<Tuple2<String, Long>> logsWithLength = Arrays.stream(logFiles)
+				.map(logFile -> Tuple2.of(logFile.getName(), logFile.length()))
+				.collect(Collectors.toList());
+			return CompletableFuture.completedFuture(logsWithLength);
+		}
+		return FutureUtils.completedExceptionally(new FlinkException("There is no log file available on the TaskExecutor."));
```

Comment:
Another reasonable behavior would be to return an empty collection. Not sure what is better here.

Reply:
It needs to confirm. It depends on what will the frontend handle the exception.

Reply:
Because there must be a log file currently, if there is no log file, it should be an exception, not empty.

Reply:
Ok, that's reasonable.